ingestion/src/metadata/ingestion/source/metabase.py

```diff
@@ -0,0 +1,234 @@
+from metadata.generated.schema.entity.data.table import Table
+from metadata.generated.schema.entity.services.dashboardService import (
+    DashboardServiceType,
+)
+from metadata.generated.schema.type.entityLineage import EntitiesEdge
+from metadata.generated.schema.type.entityReference import EntityReference
+from metadata.ingestion.api.common import Entity, WorkflowContext
+from metadata.ingestion.api.source import Source, SourceStatus
+from metadata.ingestion.models.table_metadata import Chart, Dashboard
+from metadata.ingestion.ometa.ometa_api import OpenMetadata
+from metadata.ingestion.ometa.openmetadata_rest import MetadataServerConfig
+from metadata.ingestion.source.sql_alchemy_helper import SQLSourceStatus
+from metadata.ingestion.source.sql_source import SQLConnectionConfig, SQLSourceStatus
+from metadata.utils.helpers import get_dashboard_service_or_create
+
+HEADERS = {"Content-Type": "application/json", "Accept": "*/*"}
+
+
+logger: logging.Logger = logging.getLogger(__name__)
+
+
+class MetabaseConnectionConfig(SQLConnectionConfig):
+    service_type = "MySQL"
+    session_id: str = None
+    scheme = ""
+    database_service_name: str = None
+
+    def get_connection_url(self):
+        pass
+
+
+class MetabaseSource(Source[Entity]):
+    config: MetabaseConnectionConfig
+    metadata_config: MetadataServerConfig
+    status: SQLSourceStatus
+
+    def __init__(
+        self,
+        config: MetabaseConnectionConfig,
+        metadata_config: MetadataServerConfig,
+        ctx: WorkflowContext,
+    ):
+        super().__init__(ctx)
+        self.config = config
+        self.metadata_config = metadata_config
+        self.status = SQLSourceStatus()
+        params = {}
+        params["username"] = self.config.username
+        params["password"] = self.config.password.get_secret_value()
+        resp = requests.post(
+            self.config.host_port + "/api/session/",
+            data=json.dumps(params),
+            headers=HEADERS,
+        )
+        session_id = resp.json()["id"]
+        self.metabase_session = {"X-Metabase-Session": session_id}
+        self.dashboard_service = get_dashboard_service_or_create(
+            config.service_name,
+            DashboardServiceType.Looker.name,
+            config.username,
+            config.password.get_secret_value(),
+            config.host_port,
+            metadata_config,
+        )
+        self.charts = []
+        self.metric_charts = []
+
+    @classmethod
+    def create(cls, config_dict, metadata_config_dict, ctx):
+        config = MetabaseConnectionConfig.parse_obj(config_dict)
+        metadata_config = MetadataServerConfig.parse_obj(metadata_config_dict)
+        return cls(config, metadata_config, ctx)
+
+    def next_record(self) -> Iterable[Entity]:
+        yield from self.get_dashboards()
+        yield from self.get_metrics()
+
+    def get_charts(self, charts) -> Iterable[Chart]:
```

Comment:
so we are calling get_charts from dashboards then also calling get_metrics.
isn't charts and metrics the same.